node/src/chain/consensus.rs

```diff
@@ -356,4 +350,44 @@ impl<DB: database::DB, VM: vm::VMExecution> Operations for Executor<DB, VM> {
             discarded_txs,
         })
     }
+
+    async fn add_step_elapsed_time(
+        &self,
+        round: u64,
+        step_name: StepName,
+        elapsed: Duration,
+    ) -> Result<(), Error> {
+        if let StepName::Proposal | StepName::Ratification = step_name {
+            // Later on, measurements can be collected for all steps as well
+            return Ok(());
+        }
+
+        if elapsed.as_secs() == 0 {
+            return Ok(());
+        }
```

Comment:
Why are you filtering out 0secs?

Running a `testbed.sh 30` with this codebase results in a timeout never decreased (because local nodes take less than 1 sec)

Removing it let the timeout adjust back after being increased


Skipping 0secs elapsed
```
{"timestamp":"2024-01-31T11:27:56.593875Z","level":"DEBUG","event":"execute_step","timeout":"30s","target":"dusk_consensus::phase","spans":[{"iter":0,"name":"Validation","pk":"pFPEcfxidLvwmFKR","round":76,"name":"main"}]}
{"timestamp":"2024-01-31T11:27:57.031368Z","level":"DEBUG","event":"execute_step","timeout":"20s","target":"dusk_consensus::phase","spans":[{"iter":0,"name":"Ratification","pk":"pFPEcfxidLvwmFKR","round":76,"name":"main"}]}
{"timestamp":"2024-01-31T11:27:58.714284Z","level":"DEBUG","event":"execute_step","timeout":"33s","target":"dusk_consensus::phase","spans":[{"iter":1,"name":"Proposal","pk":"pFPEcfxidLvwmFKR","round":76,"name":"main"}]}
```

Counting 0secs elapsed
```
{"timestamp":"2024-01-31T11:38:39.296578Z","level":"DEBUG","event":"execute_step","timeout":"9s","target":"dusk_consensus::phase","spans":[{"iter":1,"name":"Proposal","pk":"pFPEcfxidLvwmFKR","round":1,"name":"main"}]}
{"timestamp":"2024-01-31T11:38:48.298930Z","level":"INFO","event":"timeout-ed","target":"dusk_consensus::execution_ctx","spans":[{"iter":1,"name":"Proposal","pk":"pFPEcfxidLvwmFKR","round":1,"name":"main"}]}
{"timestamp":"2024-01-31T11:39:03.430893Z","level":"DEBUG","event":"execute_step","timeout":"12s","target":"dusk_consensus::phase","spans":[{"iter":2,"name":"Proposal","pk":"pFPEcfxidLvwmFKR","round":1,"name":"main"}]}
{"timestamp":"2024-01-31T11:39:05.517741Z","level":"DEBUG","event":"execute_step","timeout":"15s","target":"dusk_consensus::phase","spans":[{"iter":3,"name":"Proposal","pk":"pFPEcfxidLvwmFKR","round":1,"name":"main"}]}
{"timestamp":"2024-01-31T11:39:05.865108Z","level":"DEBUG","event":"execute_step","timeout":"6s","target":"dusk_consensus::phase","spans":[{"iter":0,"name":"Proposal","pk":"pFPEcfxidLvwmFKR","round":2,"name":"main"}]}
{"timestamp":"2024-01-31T11:39:09.384595Z","level":"DEBUG","event":"execute_step","timeout":"9s","target":"dusk_consensus::phase","spans":[{"iter":1,"name":"Proposal","pk":"pFPEcfxidLvwmFKR","round":2,"name":"main"}]}
{"timestamp":"2024-01-31T11:39:09.533843Z","level":"DEBUG","event":"execute_step","timeout":"6s","target":"dusk_consensus::phase","spans":[{"iter":0,"name":"Proposal","pk":"pFPEcfxidLvwmFKR","round":3,"name":"main"}]}
{"timestamp":"2024-01-31T11:39:12.757765Z","level":"DEBUG","event":"execute_step","timeout":"9s","target":"dusk_consensus::phase","spans":[{"iter":1,"name":"Proposal","pk":"pFPEcfxidLvwmFKR","round":3,"name":"main"}]}
{"timestamp":"2024-01-31T11:39:13.621173Z","level":"DEBUG","event":"execute_step","timeout":"6s","target":"dusk_consensus::phase","spans":[{"iter":0,"name":"Proposal","pk":"pFPEcfxidLvwmFKR","round":4,"name":"main"}]}
{"timestamp":"2024-01-31T11:39:16.706673Z","level":"DEBUG","event":"execute_step","timeout":"9s","target":"dusk_consensus::phase","spans":[{"iter":1,"name":"Proposal","pk":"pFPEcfxidLvwmFKR","round":4,"name":"main"}]}
```


However, I agree that we should handle this case (or more generally speaking, handle the case when avg is too low)
Probably it's worth introducing a MIN_STEP_TIMEOUT

Reply:
Indeed. A more robust approach could be to introduce a minimum step timeout and remove the check `if elapsed.as_secs()`. 

Also, it could be more accurate if we base the average value on `milliseconds` instead of seconds.

Reply:
Not sure storing elapsed times in `milliseconds` is necessary here. 
We want to give some loose margin to keep the timeout high enough for steps not to fail. 
I mean, if the elapse time is 0.5 I think it's fine to save it as 1.

Either way, we should round up when computing the adjusted timeout value.